packages/maker/src/api/defineInterface.ts

```diff
@@ -20,20 +20,36 @@ import { defineSharedPropertyType } from "./defineSpt.js";
 import type { BlueprintIcon } from "./iconNames.js";
 import type {
   InterfaceType,
+  InterfaceTypeRid,
+  InterfaceTypeStatus,
+  InterfaceTypeStatus_active,
+  InterfaceTypeStatus_experimental,
   PropertyTypeType,
   SharedPropertyType,
 } from "./types.js";
 
+/**
+ * This status indicates that the interface is reaching the end of its life and will be removed as per the
+ * deadline specified.
+ */
+export interface DeprecatedInterfaceTypeStatus {
+  message: string;
+  deadline: string;
+  replacedBy: InterfaceTypeRid | undefined;
+}
+
 export function defineInterface(
   opts: {
     apiName: string;
     displayName?: string;
     description?: string;
     icon?: { locator: BlueprintIcon; color: string };
+    status?: string | InterfaceTypeStatus;
```

Comment:
I would prefer instead of exposing the raw InterfaceTypeStatus union we flattened the union to be simply `{type:"deprecated"; message: string; deadline: string;} | {type: "active"} | {type: "experimental"}`

I think this is a nicer UX than base conjure unions.